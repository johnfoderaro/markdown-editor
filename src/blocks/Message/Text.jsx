import styled from 'styled-components';

const Text = styled.span`
  padding: 1rem;
  color: #fff;
  font-size: 1.5em;
`;

export default Text;
