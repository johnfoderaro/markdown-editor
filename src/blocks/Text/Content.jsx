import styled from 'styled-components';

const Content = styled.textarea`
  display: flex;
  flex-direction: column;
  height: 100%;
  padding: .25rem;
  border: none;
  overflow: auto;
  outline: none;
  box-shadow: none;
  resize: none;
  font-family: 'Monaco', monospace, sans-serif;
`;

export default Content;
