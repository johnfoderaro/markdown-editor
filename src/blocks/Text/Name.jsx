import styled from 'styled-components';

const Name = styled.input`
  margin-top: 1rem;
  margin-bottom: 1rem;
  padding-top: 1rem;
  padding-bottom: 1rem;
  display: flex;
  flex-direction: column;
  font-size: 2em;
  border: none;
`;

export default Name;
