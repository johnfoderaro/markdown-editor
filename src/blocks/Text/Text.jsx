import styled from 'styled-components';

import Content from './Content';
import Action from './Action';

const Text = styled.div`
  padding: 1rem;
  margin-left: auto;
  display: flex;
  flex-direction: column;
  width: 66%;
  height: 30rem;
  border: 1px solid #000;
  border-radius: .5rem;
`;

Text.Action = Action;
Text.Content = Content;

export default Text;
