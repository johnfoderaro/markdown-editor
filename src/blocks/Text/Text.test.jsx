import React from 'react';
import { mount } from 'enzyme';
import 'jest-styled-components';

import Text from './Text';

let tree;

beforeAll(() => {
  tree = mount(
    <Text>
      <Text.Name placeholder="untitled.md" value="test01.md" />
      <Text.Content placeholder="&quot;The first draft is just you telling yourself the story.&quot;" value="this is the body of test01.md" />
    </Text>,
  );
});

describe('<Text />', () => {
  it('should be defined', () => {
    expect(tree).toBeDefined();
  });
  it('should render correctly', () => {
    expect(tree).toMatchSnapshot();
  });
  it('should render children correctly', () => {
    expect(tree.find(Text.Name)).toBeDefined();
    expect(tree.find(Text.Content)).toBeDefined();
  });
});
