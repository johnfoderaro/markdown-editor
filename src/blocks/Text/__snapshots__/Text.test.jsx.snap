// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> should render correctly 1`] = `
.c1 {
  margin-top: 1rem;
  margin-bottom: 1rem;
  padding-top: 1rem;
  padding-bottom: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 2em;
  border: none;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  padding: .25rem;
  border: none;
  overflow: auto;
  outline: none;
  box-shadow: none;
  resize: none;
  font-family: 'Monaco',monospace,sans-serif;
}

.c0 {
  padding-right: 1rem;
  padding-left: 1rem;
  height: 100%;
  width: 80%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<styled.div>
  <StyledComponent
    forwardedClass={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "Content": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "lastClassName": "c2",
            "rules": Array [
              "
  display: flex;
  flex-direction: column;
  height: 100%;
  padding: .25rem;
  border: none;
  overflow: auto;
  outline: none;
  box-shadow: none;
  resize: none;
  font-family: 'Monaco', monospace, sans-serif;
",
            ],
          },
          "displayName": "styled.textarea",
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "textarea",
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
        "Name": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "c1",
            "rules": Array [
              "
  margin-top: 1rem;
  margin-bottom: 1rem;
  padding-top: 1rem;
  padding-bottom: 1rem;
  display: flex;
  flex-direction: column;
  font-size: 2em;
  border: none;
",
            ],
          },
          "displayName": "styled.input",
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "input",
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-htpNat",
          "isStatic": true,
          "lastClassName": "c0",
          "rules": Array [
            "
  padding-right: 1rem;
  padding-left: 1rem;
  height: 100%;
  width: 80%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-htpNat",
        "target": "div",
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="c0"
    >
      <styled.input
        placeholder="untitled.md"
        value="test01.md"
      >
        <StyledComponent
          forwardedClass={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": undefined,
              "componentStyle": ComponentStyle {
                "componentId": "sc-bdVaJa",
                "isStatic": true,
                "lastClassName": "c1",
                "rules": Array [
                  "
  margin-top: 1rem;
  margin-bottom: 1rem;
  padding-top: 1rem;
  padding-bottom: 1rem;
  display: flex;
  flex-direction: column;
  font-size: 2em;
  border: none;
",
                ],
              },
              "displayName": "styled.input",
              "render": [Function],
              "styledComponentId": "sc-bdVaJa",
              "target": "input",
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
          placeholder="untitled.md"
          value="test01.md"
        >
          <input
            className="c1"
            placeholder="untitled.md"
            value="test01.md"
          />
        </StyledComponent>
      </styled.input>
      <styled.textarea
        placeholder="\\"The first draft is just you telling yourself the story.\\""
        value="this is the body of test01.md"
      >
        <StyledComponent
          forwardedClass={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": undefined,
              "componentStyle": ComponentStyle {
                "componentId": "sc-bwzfXH",
                "isStatic": true,
                "lastClassName": "c2",
                "rules": Array [
                  "
  display: flex;
  flex-direction: column;
  height: 100%;
  padding: .25rem;
  border: none;
  overflow: auto;
  outline: none;
  box-shadow: none;
  resize: none;
  font-family: 'Monaco', monospace, sans-serif;
",
                ],
              },
              "displayName": "styled.textarea",
              "render": [Function],
              "styledComponentId": "sc-bwzfXH",
              "target": "textarea",
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
          placeholder="\\"The first draft is just you telling yourself the story.\\""
          value="this is the body of test01.md"
        >
          <textarea
            className="c2"
            placeholder="\\"The first draft is just you telling yourself the story.\\""
            value="this is the body of test01.md"
          />
        </StyledComponent>
      </styled.textarea>
    </div>
  </StyledComponent>
</styled.div>
`;
